docs/design/da-pki-revocation-design.md

```diff
@@ -0,0 +1,213 @@
+- DCL checks that `CRLSignerCertificate` is either a valid PAI (signed by a trusted PAA from DCL) or a PAA present on the ledger.
+- It can be complicated to create an interactive PoP of `CRLSignerCertificate` for some PAIs.
+- Indirect CRLs are out of scope for now
+
+## Transactions
+
+### 1. ADD_PKI_REVOCATION_DISTRIBUTION_POINT
+Publishes a PKI Revocation distribution endpoint owned by the Vendor. 
+
+If `crlSignerCertificate` is a PAA (root certificate), then it must be present on DCL.
+
+If `crlSignerCertificate` is a PAI (intermediate certificate), then it must be chained back to a valid PAA (root certificate) present on DCL.
+In this case `crlSignerCertificate` is not required to be present on DCL, and will not be added to DCL as a result of this transaction.
+If PAI needs to be added to DCL, it should be done via `ADD_X509_CERT` transaction.
+
+Publishing the revocation distribution endpoint doesn't automatically remove PAI (Intermediate certificates)
+and DACs (leaf certificates) added to DCL if they are revoked in the CRL identified by this distribution point.
+
+
+- Who can send:
+    - Vendor account
+    - `vid` field in the transaction (`VendorID`) must be equal to the Vendor account's VID
+    - `vid` field in the `CRLSignerCertificate` (for vendor-scoped PAAs and PAIs) must be equal to the Vendor account's VID
+- Parameters:
+    - vid: `uint16` -  Vendor ID (positive non-zero)
+    - pid: `optional(uint16)` -  Product ID (positive non-zero)
+    - isPAA: `bool` -  True if the revocation information distribution point relates to a PAA
+    - label: `string` -  A label to disambiguate multiple revocation information partitions of a particular issuer.
+    - crlSignerCertificate: `string` -  PEM encoded certificate (string or path to file containing data)
+    - issuerSubjectKeyID: `string` -  crlSignerCertificate's `Subject Key Id` in hex string format, e.g: `5A:88:0E:6C:36:53:D0:7F:B0:89:71:A3:F4:73:79:09:30:E6:2B:DB`
+    - dataUrl: `string` -  The URL where to obtain the information in the format indicated by the RevocationType field
+    - dataFileSize: `optional(uint64)` -  Total size in bytes of the file found at the DataUrl. Must be omitted if RevocationType is 1.
+    - dataDigest: `optional(string)` -  Digest of the entire contents of the associated file downloaded from the DataUrl. Must be omitted if RevocationType is 1.
+    - dataDigestType: `optional(uint32)` - The type of digest used in the DataDigest field from the list of [1, 7, 8, 10, 11, 12] (IANA Named Information Hash Algorithm Registry).
+    - revocationType: `uint32` - The type of file found at the DataUrl for this entry. Supported types: 1 - RFC5280 Certificate Revocation List (CRL).
+- In State:
+    - `pki/RevocationDistributionPoint/value/<IssuerSubjectKeyID>` -> list of Revocation Distribution Points
+    - `pki/RevocationDistributionPoint/value/<IssuerSubjectKeyID>/<vid>/<label>`-> Revocation Distribution Point
```

Comment:
Why is the vid included in this query?

You could not have the same IssuerSubjectKeyID for a different vid.

Reply:
The [spec](https://github.com/CHIP-Specifications/connectedhomeip-spec/blob/master/src/service_device_management/DistributedComplianceLedger.adoc#84-label) says that `Uniqueness within the Device Attestation PKI Revocation Distribution Points schema SHALL be enforced against the tuple containing all of: (VendorID, Label, IssuerSubjectKeyID)`.

Right now we don't have any specific validation rules that that `You could not have the same IssuerSubjectKeyID for a different vid`. It seems usually it's the case because we check that PAIs are chained back to valid PAAs, so it's unlikely that there two valid PAIs with the same `IssuerSubjectKeyID` and different `vid`.
But for non-VID scoped PAA it can be the case (see https://github.com/zigbee-alliance/distributed-compliance-ledger/pull/472/files#r1186284795, that any Vendor can propose a Revocation endpoint for non-vid scoped PAAs, as was agreed on DCL TT call).